server/game/gameSystems/DiscardSpecificCardSystem.ts

```diff
@@ -0,0 +1,34 @@
+import type { AbilityContext } from '../core/ability/AbilityContext';
+import { Card } from '../core/card/Card';
+import { EventName, Location } from '../core/Constants';
+import { GameEvent } from '../core/event/GameEvent';
+import { CardTargetSystem, ICardTargetSystemProperties } from '../core/gameSystem/CardTargetSystem';
+
+// eslint-disable-next-line @typescript-eslint/no-empty-object-type
+export interface IDiscardSpecificCardProperties extends ICardTargetSystemProperties {}
+
+export class DiscardSpecificCardSystem<TContext extends AbilityContext = AbilityContext> extends CardTargetSystem<TContext, IDiscardSpecificCardProperties> {
+    public override readonly name = 'discardSpecificCard';
+    public override readonly eventName = EventName.OnCardsDiscarded;
+
+    public override canAffect(card: Card, context: TContext, additionalProperties: Record<string, any> = {}): boolean {
+        return card.location !== Location.Discard && super.canAffect(card, context, additionalProperties);
+    }
+
+    public override getEffectMessage(context: TContext): [string, any[]] {
+        const properties = this.generatePropertiesFromContext(context);
+        return [
+            'discard {0}',
+            [properties.target]
+        ];
+    }
+
+    protected override addPropertiesToEvent(event, card: Card, context: TContext, additionalProperties: Record<string, any> = {}): void {
+        event.discardedFromLocation = card.location;
+        super.addPropertiesToEvent(event, card, context, additionalProperties);
+    }
+
+    public eventHandler(event): void {
```

Comment:
nit: sorry I didn't catch this before, by convention we're putting this as the first method in the class since it's usually the most important